codec-http/src/main/java/io/netty/handler/codec/http/HttpRequestEncoder.java

```diff
@@ -38,7 +39,7 @@ public boolean acceptOutboundMessage(Object msg) throws Exception {
     @Override
     protected void encodeInitialLine(ByteBuf buf, HttpRequest request) throws Exception {
         AsciiString method = request.method().name();
-        buf.writeBytes(method.array(), method.arrayOffset(), method.length());
```

Comment:
@trustin @normanmaurer - Note that I got rid of the `arrayOffset()` method to simplify the interface a bit. I don't think this is a problem if when creating a `ByteString` a copy operation is done....however if the intention is to not copy (share the underlying array) we won't be able to do that effectively without `arrayOffset()`.

Should I restore `arrayOffset()` and allow "no copy of sub array" operations to take place?  I can submit a followup PR perhaps...I think a few areas will need investigation if we support `arrayOffset()`.